src/node.cc

```diff
@@ -4202,8 +4211,10 @@ void Init(int* argc,
 #endif
 
   // Allow for environment set preserving symlinks.
-  if (auto preserve_symlinks = secure_getenv("NODE_PRESERVE_SYMLINKS")) {
-    config_preserve_symlinks = (*preserve_symlinks == '1');
+  {
+    std::string text;
+    config_preserve_symlinks =
+        secure_getenv("NODE_PRESERVE_SYMLINKS", &text) && *text.c_str() == '1';
```

Comment:
`(text.length() == 1 && text[0] == '1')`? Right now, it appears a value of ``10` or `100` would also be accepted, in violation of the docs. Though maybe that's a bug for another PR?